[expr] Replace 'could' and 'might' (#4364)
cplusplus/draft: source/expressions.tex

\pnum
\begin{note}
Historically, lvalues and rvalues were so-called

> because they can appear on the left- and right-hand side of an assignment

Reviewer: This is awkward, given the parenthetical. Happy to turn a blind eye, but not for 23.

Reviewer 2: Looks like github lost my comment on this :(
```suggestion
because they appeared on the left- and right-hand side of an assignment
```

can be converted to type \tcode{T2}
if the cv-combined type of \tcode{T1} and \tcode{T2} is \tcode{T2}.
\begin{note}
If a program were able to assign a pointer of type \tcode{T**} to a pointer of
type \tcode{const} \tcode{T**} (that is, if line \#1 below were

> allowed), a program would be able to inadvertently modify a const object

Reviewer 2: ```suggestion
allowed), it would be possible for a const object to be inadvertently modified
```
I think we want to say that it's possible for this to happen, not that a program is able to do it. I still think this is a wording regression, since it creates the impression that this rule prevents inadvertent modification of const objects in general, which it does not.

> (as it is done on line \#2). For example,

Reviewer 2: ```suggestion
(as is done on line \#2). For example,
```

operators otherwise\iref{class.copy.assign}.
\begin{note}
These special member functions are implicitly defined as

> usual, and can therefore be defined as deleted.

Reviewer: This sounds uncomfortably ambiguous with the meaning of "and therefore you the user can define it as deleted". The previous wording wasn't quite so readily interpretable that way. Maybe extra words like "and can therefor end up as being defined as deleted". (Maybe for 23.)

Reviewer 2: How about:
```suggestion
usual, which can result in them being defined as deleted.
```

Reviewer: Great, another example to go on the "noteworthy examples" list!

  auto g2 = [=](auto a) {
    int selector[sizeof(a) == 1 ? 1 : 2]{};
    f(x, selector);             // OK: captures \tcode{x}, can call \#1 or \#2
  };

  auto g3 = [=](auto a) {
    typeid(a + x);              // captures \tcode{x} regardless of whether \tcode{a + x} is an unevaluated operand
  };
}
\end{codeblock}
Within \tcode{g1}, an implementation can optimize away
the capture of \tcode{x} as it is not odr-used.
\end{example}
\begin{note}
The set of captured entities is determined syntactically,

> and entities can be implicitly captured

Reviewer 2: ```suggestion
and entities are implicitly captured
```
